packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1153,7 +1173,11 @@ function commitUnmount(
           do {
             const {destroy, tag} = effect;
             if (destroy !== undefined) {
-              if ((tag & HookLayout) !== NoHookEffect) {
+              if (
+                // TODO: WAT
```

Comment:
?

Reply:
This was a note to follow-up to see if this behavior as ok.

As written, during unmount this will interleave destroy functions for insertion and layout effects, as:

```
Destroy Insertion 1 for Component A
Destroy Insertion 2 for Component A
Destroy Layout 1 for Component A
Destroy Layout 2 for Component A
Destroy Insertion 1 for Component B
Destroy Insertion 2 for Component B
Destroy Layout 1 for Component B
Destroy Layout 2 for Component B
```